### PR TITLE
feat(apig): add apig endpoint service whitelist resource support

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
add apig endpoint service whitelist resource support
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add apig endpoint service whitelist resource support
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/apig" TESTARGS="-run TestAccEndpointWhiteList_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run TestAccEndpointWhiteList_basic -timeout 360m -parallel 4
=== RUN   TestAccEndpointWhiteList_basic
=== PAUSE TestAccEndpointWhiteList_basic
=== CONT  TestAccEndpointWhiteList_basic
--- PASS: TestAccEndpointWhiteList_basic (427.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      427.398s

```

### CODE REVIEW

File: docs/resources/apig_endpoint_whitelist.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+* `id` - The resource ID (also instance ID).
+
+## Import
+
+endpoint service whitelists can be imported using their `id` and the ID of the related dedicated instance,
+  separated by a slash, e.g.
+
+```shell
+$ terraform import huaweicloud_apig_endpoint_whitelist.test <instance_id>/<id>
+```
```
Comment (Lance52259) [inline docs/resources/apig_endpoint_whitelist.md]:
```suggestion
Endpoint service whitelists can be imported using their `id`, e.g.

```shell
$ terraform import huaweicloud_apig_endpoint_whitelist.test <id>
```
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+)
+
+func getEndpointWhiteListFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.ApigV2Client(acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating APIG v2 client: %s", err)
+	}
+
+	opts := endpoints.ListOpts{
+		InstanceId: state.Primary.Attributes["instance_id"],
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]:
```suggestion
		InstanceId: state.Primary.ID,
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_whitelist.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	importedId := d.Id()
+	parts := strings.SplitN(importedId, "/", 2)
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("invalid format specified for import ID, want '<instance_id>/<id>', "+
+			"but got '%s'", importedId)
+	}
+
+	d.SetId(parts[1])
+	return []*schema.ResourceData{d}, d.Set("instance_id", parts[0])
+}
```
Comment (Lance52259) [inline huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_whitelist.go:225]: Useless function, the instanceId already as the resource ID.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+					resource.TestCheckResourceAttrPair(rName, "instance_id",
+						"huaweicloud_apig_instance.test", "id"),
+					resource.TestCheckOutput("is_update_successful", "true"),
+				),
+			},
+			{
+				ResourceName:      rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccEndpointWhitelistImportStateFunc(rName),
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]:
```suggestion
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+func testAccEndpointWhiteList_update(rName string) string {
+	return fmt.Sprintf(`
+%[1]s
+
+resource "huaweicloud_apig_endpoint_whitelist" "test" {
+  instance_id = huaweicloud_apig_instance.test.id
+  whitelists  = [
+    "iam:domain::3cc2018e40394f7c9692f1713e76234d",
+	"iam:domain::4cc2018e40394f7c9692f1713e76234d",
+    "iam:domain::5cc2018e40394f7c9692f1713e76234d",
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]: wrong format.

File: huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_whitelist.go (hunk 3 above)
Comment (Lance52259) [inline huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_whitelist.go]:
```suggestion
func resourceEndpointWhiteListImportState(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData,
	error) {
	return []*schema.ResourceData{d}, d.Set("instance_id", d.Id())
}
```

File: docs/resources/apig_endpoint_whitelist.md (hunk 1 above)
Comment (jinyangyang222) [reply docs/resources/apig_endpoint_whitelist.md]: fixed.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go (hunk 2 above)
Comment (jinyangyang222) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]: fixed.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go (hunk 4 above)
Comment (jinyangyang222) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]: fixed.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go (hunk 5 above)
Comment (jinyangyang222) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]: fixed.

File: huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_whitelist.go (hunk 3 above)
Comment (jinyangyang222) [reply huaweicloud/services/apig/resource_huaweicloud_apig_endpoint_whitelist.go]: fixed.

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  instance_id = huaweicloud_apig_instance.test.id
+  whitelists  = [
+    "iam:domain::1cc2018e40394f7c9692f1713e76234d",
+    "iam:domain::2cc2018e40394f7c9692f1713e76234d",
+  ]
+}
+
+output is_add_successful {
+  value = length(huaweicloud_apig_endpoint_whitelist.test.whitelists) == 2
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]:
```
resource.TestCheckResourceAttr(rName, "whitelists.#", "2"),
```

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+		InstanceId: state.Primary.ID,
+	}
+	resp, err := endpoints.ListPermissions(client, opts)
+	if err != nil {
+		return nil, err
+	}
+	if len(resp) < 1 {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return resp, nil
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]:
```suggestion
	resp, err := endpoints.ListPermissions(client, opts)
	if err == nil && len(resp) < 1 {
		return nil, golangsdk.ErrDefault404{}
	}
	return resp, err
```

Comment (jinyangyang222) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_endpoint_whitelist_test.go]: fixed.

**Pull request merged**